llvm-runtimes/libgcc/libgcc-19.1.7-r1.ebuild

```
@@ -0,0 +1,178 @@
+	# visibility of all symbols.
+
+	llvm_prepend_path "${LLVM_MAJOR}"
+
+	# LLVM_ENABLE_ASSERTIONS=NO does not guarantee this for us, #614844
+	use debug || local -x CPPFLAGS="${CPPFLAGS} -DNDEBUG"
+
+	export CC=${CTARGET}-clang CXX=${CTARGET}-clang++
+	strip-unsupported-flags
+
+	local mycmakeargs=(
+		-DCOMPILER_RT_INSTALL_PATH="${EPREFIX}/usr/lib/clang/${LLVM_MAJOR}"
+
+		-DCOMPILER_RT_INCLUDE_TESTS=$(usex test)
+		-DCOMPILER_RT_BUILD_CRT=OFF
+		-DCOMPILER_RT_BUILD_CTX_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_LIBFUZZER=OFF
+		-DCOMPILER_RT_BUILD_MEMPROF=OFF
+		-DCOMPILER_RT_BUILD_ORC=OFF
+		-DCOMPILER_RT_BUILD_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_SANITIZERS=OFF
+		-DCOMPILER_RT_BUILD_XRAY=OFF
+
+		-DCOMPILER_RT_BUILTINS_HIDE_SYMBOLS=OFF
+
+		-DPython3_EXECUTABLE="${PYTHON}"
+	)
+
+	# disable building non-native runtimes since we don't do multilib
+	if use amd64; then
+		mycmakeargs+=(
+			-DCAN_TARGET_i386=OFF
+		)
+	fi
+
+	if use test; then
+		mycmakeargs+=(
+			-DLLVM_EXTERNAL_LIT="${EPREFIX}/usr/bin/lit"
+			-DLLVM_LIT_ARGS="$(get_lit_flags)"
+
+			-DCOMPILER_RT_TEST_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang"
+			-DCOMPILER_RT_TEST_CXX_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang++"
+		)
+	fi
+
+	cmake_src_configure
+}
+
+# Usage: deps
+gen_ldscript() {
+	local output_format
+	output_format=$($(tc-getCC) ${CFLAGS} ${LDFLAGS} -Wl,--verbose 2>&1 | sed -n 's/^OUTPUT_FORMAT("\([^"]*\)",.*/\1/p')
+	[[ -n ${output_format} ]] && output_format="OUTPUT_FORMAT ( ${output_format} )"
+
+	cat <<-END_LDSCRIPT
+/* GNU ld script
+   Include missing dependencies
+*/
+${output_format}
+GROUP ( $@ )
+END_LDSCRIPT
+}
+
+src_compile() {
+	cmake_src_compile
+
+	local rtlib=$(
+		"${CC}" -rtlib=compiler-rt -resource-dir="${BUILD_DIR}" \
+			-print-libgcc-file-name || die
+	)
+
+	# Use the llvm-libgcc's version script to produce libgcc.{a,so}, which
+	# combines compiler-rt and libunwind into a libgcc replacement.
+	#
+	# What we do here is similar to what upstream does[0], with the following
+	# differences:
+	#
+	# * We build the local copy of compiler-rt manually, to have a full control
+	#   over CMake options.
+	# * Upstream links the locally built copy of libunwind statically. We link the
+	#   system-wide libunwind dynamically.
+	#
+	# [0] https://github.com/llvm/llvm-project/blob/llvmorg-19.1.7/llvm-libgcc/CMakeLists.txt#L102-L120
+	"${CC}" -E -xc \
+		"${WORKDIR}/llvm-libgcc/gcc_s.ver.in" \
+		-o gcc_s.ver || die
+	"${CC}" -nostdlib \
+		${LDFLAGS} \
+		-Wl,--version-script,gcc_s.ver \
+		-Wl,--undefined-version \
+		-Wl,--whole-archive \
+		"${rtlib}" \
+		-Wl,-soname,libgcc_s.so.1.0 \
+		-lc -lunwind -shared \
+		-o libgcc_s.so.1.0 || die
+	# Generate libgcc_s.so ldscript for inclusion of libunwind as a
+	# dependency so that `clang -lgcc_s` works out of the box.
+	local deps=(
+		libgcc_s.so.1.0
+		libunwind.so.1.0
+	)
+	gen_ldscript "${deps[*]}" > libgcc_s.so || die
```

```suggestion
	gen_ldscript libgcc_s.so.1.0 libunwind.so.1.0 > libgcc_s.so || die
```

The indirection seems unnecessary, and the way you're passing the arguments is just weird.